torch/csrc/jit/codegen/cuda/arith.cpp

```diff
@@ -1263,21 +1267,68 @@ std::vector<Int*> convertToIntVector(const std::vector<int>& x) {
 TensorView* gather(
     TensorView* inp,
     const std::vector<int>& window_shape,
-    const std::vector<std::vector<int>>& pad_width) {
+    const std::vector<std::vector<int>>& pad_width,
+    const std::vector<int>& strides) {
   std::vector<Int*> window_shape_int = convertToIntVector(window_shape);
   std::vector<std::vector<Int*>> pad_width_int;
   std::transform(
       pad_width.begin(),
       pad_width.end(),
       std::back_inserter(pad_width_int),
       [](const std::vector<int>& x) { return convertToIntVector(x); });
-  return gather(inp, window_shape_int, pad_width_int);
+  return gather(inp, window_shape_int, pad_width_int, strides);
+}
+
+namespace {
+
+// Return a new TensorDomain with given root domains. Apply strides if
+// necessary. With non-unit strides, strided domains become an rfactor
+// domain.
+TensorDomain* generateTensorDomainWithStrides(
+    const std::vector<IterDomain*>& root_domains,
+    const std::vector<int>& strides) {
+  std::vector<IterDomain*> strided_domains;
+
+  // If strides are just unit strides, don't apply striding
+  if (strides.empty() || std::all_of(strides.begin(), strides.end(), [](int s) {
+        return s == 1;
+      })) {
+    return new TensorDomain(
+        root_domains, std::vector<bool>(root_domains.size(), true));
+  }
+
+  for (const auto i : c10::irange(root_domains.size())) {
+    auto root_dom = root_domains.at(i);
+
+    if (i >= strides.size() || strides[i] == 1) {
+      strided_domains.push_back(root_dom);
+      continue;
+    }
+
+    // Split the root domain by the stride
+    auto split_out = root_dom->stridedSplit(strides[i]);
+    strided_domains.push_back(split_out.first);
+    strided_domains.push_back(split_out.second);
```

Comment:
Do you it might be nicer on the user side to push strided dimensions to the end of the tensor instead of interleaving them?

Reply:
It might be, but it might also make sense not to deviate much from the normal split operation. Not really sure.